react-native/react/reducers/index.js

```diff
@@ -1,16 +1,39 @@
 'use strict'
+/* @flow */
 
 import { combineReducers } from 'redux'
 import login from './login'
+// $FlowFixMe login2 isnt typed
 import login2 from './login2'
 import devices from './devices'
 import search from './search'
 import profile from './profile'
 import config from './config'
 import tabbedRouter from './tabbed-router.js'
+import {List} from 'immutable'
 
-export default function (state, action) {
-  return combineReducers({
+let history = List()
+let index = 0
+
+type State = { [key: string]: any }
+
+function timeTravel (state: State, action: any): State {
+  if (action.type !== 'timetravel') {
+    history = history.slice(0, index + 1).push(state)
+    index = history.size - 1
+    return state
+  } else {
+    const { direction } = action.payload
+    if (direction === 'back') {
+      return history.get(--index, state)
+    } else {
+      return history.get(++index, state)
+    }
+  }
+}
+
+export default function (state: State, action: any): State {
+  const nextState = combineReducers({
```

Comment:
This should actually just be `export default combineReduers({ … })`, it's wasteful to rebuild the reducer every time. This isn't actually part of your change, but is relevant to the below.